api/repositories/build_repository_test.go

```diff
@@ -511,6 +511,64 @@ var _ = Describe("BuildRepository", func() {
 			})
 		})
 	})
+
+	Describe("ListBuilds", func() {
+		var (
+			app1GUID     string
+			package1GUID string
+			namespace1   *korifiv1alpha1.CFSpace
+			cfOrg        *korifiv1alpha1.CFOrg
+			build1GUID   string
+			build1       *korifiv1alpha1.CFBuild
+			buildRecords []repositories.BuildRecord
+			fetchError   error
+		)
+
+		BeforeEach(func() {
+			build1GUID = uuid.NewString()
+			app1GUID = uuid.NewString()
+			package1GUID = uuid.NewString()
+			cfOrg = createOrgWithCleanup(ctx, prefixedGUID("org"))
+			namespace1 = createSpaceWithCleanup(ctx, cfOrg.Name, prefixedGUID("space2"))
+			build1 = &korifiv1alpha1.CFBuild{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      build1GUID,
+					Namespace: namespace1.Name,
+					Labels: map[string]string{
+						korifiv1alpha1.SpaceGUIDKey: namespace1.Name,
+					},
+				},
+				Spec: korifiv1alpha1.CFBuildSpec{
+					PackageRef: corev1.LocalObjectReference{
+						Name: package1GUID,
+					},
+					AppRef: corev1.LocalObjectReference{
+						Name: app1GUID,
+					},
+					Lifecycle: korifiv1alpha1.Lifecycle{
+						Type: "buildpack",
+						Data: korifiv1alpha1.LifecycleData{
+							Buildpacks: []string{},
+							Stack:      "",
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, build1)).To(Succeed())
+		})
+
+		It("if user is a Space Developer it returns a list with one element", func() {
```

Comment:
That `It` implicitly changes the test fixture by creating a role binding. Therefore, I would suggest

```
Describe("ListBuilds", func() {
  BeforeEach() // create two spaces, one build in space
  It("returns an empty array")

  When("the user is a space developer"...
    BeforeEach(....) // create a role binding for ONE of the spaces here
    It("returns builds from authorised spaces" // check you only get the build from the authorised space
```